src/components/secrets/SecretList.js

```diff
@@ -52,9 +60,31 @@ class SecretList extends Component {
       'i'
     );
 
-    const secrets = this.props.secrets
-      .filter(secret => fuzzyRegexp.test(secret.title))
-      .sortBy(secret => secret.get('title').toLowerCase());
+    const special = this.props.showAll || this.props.showMine || this.props.showShared;
+
+    let secrets = this.props.secrets
+      .filter(secret => fuzzyRegexp.test(secret.title));
+
+    let folders = new Immutable.Map();
+
+    if (special) {
+      secrets.forEach((secret) => {
+        const foldersSeq = secret.getIn(['users', secretin.currentUser.username, 'folders']).entrySeq();
```

Comment:
Please avoid using secretin in components. The cuttentUser is already in `AppUIStore`